src/encoding.ts

```diff
@@ -130,21 +131,21 @@ export interface Encoding<F extends Field> {
    * _Note:_ When using `stroke` channel, `color ` channel will be ignored. To customize both stroke and fill, please use `stroke` and `fill` channels (not `stroke` and `color`).
    */
 
-  stroke?: ColorFieldDefWithCondition<F> | ColorValueDefWithCondition<F> | GradientValueDef<F>;
+  stroke?: ColorFieldDefWithCondition<F> | ColorValueDefWithCondition<F> | Gradient;
```

Comment:
Not sure why you need `| Gradient` here given you already added it to colorvaluedef  already

Reply:
Yeah that was me rebasing to master yesterday haha :D I fixed following your instructions but it still fail to compile during the parsing phase since SignalRef expects signal is String not Gradient. I think it was where I declare the Gradient from vega-typings, should it be directly GradientLinear or should it be a string?